samples/algorithms/sudoku-grover/Program.cs

```diff
@@ -0,0 +1,474 @@
+                        for (int iSub = iSubGridStart; iSub < i; iSub++)
+                        {
+                            for (int jSub = jSubGridStart; jSub < j; jSub++)
+                            {
+                                if (puzzle[iSub, jSub] != 0)
+                                    startingNumberConstraints.Add(ValueTuple.Create(emptyIndex, puzzle[iSub, jSub] - 1));
+                                else if (iSub < i && jSub < j)
+                                    emptySquareEdges.Add(ValueTuple.Create(emptyIndex, emptyIndexes[iSub, jSub]));
+                            }
+                        }
+                        for (int ii = 0; ii < size; ii++)
+                        {
+                            if (puzzle[ii, j] != 0)
+                                startingNumberConstraints.Add(ValueTuple.Create(emptyIndex, puzzle[ii, j] - 1));
+                            else if (ii < i)
+                                emptySquareEdges.Add(ValueTuple.Create(emptyIndex, emptyIndexes[ii, j]));
+                        }
+                        for (int jj = 0; jj < size; jj++)
+                        {
+                            if (puzzle[i, jj] != 0)
+                                startingNumberConstraints.Add(ValueTuple.Create(emptyIndex, puzzle[i, jj] - 1));
+                            else if (jj < j)
+                                emptySquareEdges.Add(ValueTuple.Create(emptyIndex, emptyIndexes[i, jj]));
+                        }
+                    }
+                }
+            }
+        }
+
+        /// # Summary
+        /// Pause execution with a message and wait for a key to be pressed to continue
+        public static void Pause()
+        {
+            System.Console.WriteLine("\n\nPress any key to continue...\n\n");
+            System.Console.ReadKey();
+        }
+
+        /// # Summary
+        /// Classical Sudoku solution using Recursive Depth First Search 
+        /// ## puzzle
+        /// The Sudoku puzzle
+        /// ## size
+        /// The size of the puzzle i.e. 4 or 9
+        /// ## subSize
+        /// The size of the subGrids i.e. if size = 9, subSize = 3
+        static bool SolveSudukoClassic(int[,] puzzle, int size, int subSize)
+        {
+            // find empty cell will least possible options and try each
+            Candidate emptyCell = BestSquare(puzzle, size, subSize);
+            if (emptyCell == null)
+                return true; // no more empty cells --- success!!
+            if (emptyCell.possibleValues.Count == 0)
+                return false; // there's an empty cell, but no possible values -- dead end
+            foreach (int possibleValue in emptyCell.possibleValues)
+            {
+                puzzle[emptyCell.i, emptyCell.j] = possibleValue;
+                bool result = SolveSudukoClassic(puzzle, size, subSize);
+                if (result)
+                    return true;
```

Comment:
The `result` variable isn't used elsewhere, such that it's helpful to just return directly.
```suggestion
                if (SolveSudukoClassic(puzzle, size, subSize)) return true;
```